src/main/java/com/example/onlinebookstore/service/UserServiceImpl.java

```diff
@@ -31,6 +35,13 @@ public UserDto registerUser(UserRequestDto userRequestDto) throws RegistrationEx
         Role defaultRole = roleRepository.findByRole(Role.RoleName.USER);
         user.setRoles(Collections.singleton(defaultRole));
         User savedUser = userRepository.save(user);
+        shoppingCartRepository.save(new ShoppingCart(user));
```

Comment:
You can add method to shopping cart service for creating shhopping cart for user. But maybe this ok